### PR TITLE
feat: adds strip runtime

### PR DESCRIPTION
**Pull request by madou**
Breaking down the CSS extract spike into smaller PRs.

**TODO**

- [x] Edge cases
- [x] Test edge cases
- [x] Clean up types

### CODE REVIEW

File: packages/babel-plugin-strip-runtime/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "url": "https://github.com/atlassian-labs/compiled.git",
+    "directory": "packages/babel-plugin-strip-runtime"
+  },
+  "main": "./dist/index.js",
+  "module": "./dist/index.js",
+  "types": "./dist/index.d.ts",
+  "files": [
+    "dist",
+    "src",
+    "README.md"
```
Comment (danieldelcore) [inline packages/babel-plugin-strip-runtime/package.json:20]: Is readme here intentional?

File: packages/babel-plugin-strip-runtime/src/__tests__/first-party-strip.test.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  });
+
+  if (!result?.code) {
+    throw new Error();
+  }
+
+  return result.code;
+};
+
+describe('first party strip runtime', () => {
```
Comment (danieldelcore) [inline packages/babel-plugin-strip-runtime/src/__tests__/first-party-strip.test.tsx:38]: partly? or party 🥳 ?

File: packages/babel-plugin-strip-runtime/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        const callee = path.node.callee;
+        if (isCreateElement(callee)) {
+          // We've found something that looks like React.createElement(...)
+          // Now we want to check if it's from the Compiled Runtime and if it is - replace with its children.
+          const component = path.node.arguments[0];
+          if (!isCCComponent(component)) {
+            return;
+          }
+
+          const [, , compiledStyles, nodeToReplace] = path.get('arguments');
```
Comment (danieldelcore) [inline packages/babel-plugin-strip-runtime/src/index.tsx:53]: Nit:

`[, ,` this is new to me, I'm assuming we're ignoring the first two params? 

Maybe a prefixing them with a `_` would make this seem a little less weird. 

`[_thing, _otherThing, ....`

**Review by danieldelcore — APPROVED** [review]
LGTM!!! Had some questions / suggestions but otherwise G2G! NICE

File: packages/babel-plugin-strip-runtime/src/__tests__/first-party-strip.test.tsx (hunk 2 above)
Comment (madou) [reply packages/babel-plugin-strip-runtime/src/__tests__/first-party-strip.test.tsx:38]: yes party :)

File: packages/babel-plugin-strip-runtime/package.json (hunk 1 above)
Comment (madou) [reply packages/babel-plugin-strip-runtime/package.json:20]: to include? yeah

whats ur thoughts
